src/acomms/modemdriver/popoto_driver.cpp

```diff
@@ -268,74 +287,65 @@ void goby::acomms::popotoDriver::send_range_request(int dest)
 
     // Send over the wire
     signal_and_write(range.str());
-
 }
 
 // --------------------------- Incoming msgs ------------------------------------------------
-void goby::acomms::popotoDriver::do_work()
+void goby::acomms::PopotoDriver::do_work()
 {
     std::string in;
     while (modem_read(&in))
     {
         try
         {
-            // let others know about the raw feed. Can remove this if not needed
-            protobuf::ModemRaw raw;
-            raw.set_raw(in);
-            ModemDriverBase::signal_raw_incoming(raw);
-
             // Remove VT100 sequences (if they exist) and popoto prompt
-            std::cerr << "This is then recieved string: " << in << std::endl;
             in = StripString(in, "Popoto->");
+            constexpr const char *VT100_BOLD_ON = "\x1b[1m", *VT100_BOLD_OFF = "\x1b[0m";
+
             in = StripString(in, VT100_BOLD_ON);
             in = StripString(in, VT100_BOLD_OFF);
 
-            std::string binMsg;
-            if(json::accept(in))
-            {
-                binMsg = ProcessJSON(in);
-                protobuf::ModemTransmission msg;
-                if (!binMsg.empty()){
+            protobuf::ModemRaw raw;
+            raw.set_raw(in);
+            ModemDriverBase::signal_raw_incoming(raw);
```

Comment:
I think all the incoming messages serial messages are captured here? It seemed reasonable to me to have this in one place, after stripping the escape codes and "Popoto->" prompt? Let me know if this isn't sufficient for some reason

Reply:
Yeah, this looks good to me.